source/crud/crud.rst

```diff
@@ -1129,148 +1237,183 @@ Find And Modify
      *
-     * @see http://docs.mongodb.org/manual/reference/command/findAndModify/
+     * This option is sent only if the caller explicitly provides a value. The default is to not send a value.
+     *     
+     * @see https://docs.mongodb.org/manual/reference/command/findAndModify/
      */
     upsert: Optional<Boolean>;
-
-    /**
-     * Optionally specifies a collation to use in MongoDB 3.4 and higher.
-     *
-     * The default is no value: the driver sends no "collation" document to the
-     * server with the "findAndModify" command.
-     */
-    collation: Optional<Document>;
   }
 
   class FindOneAndUpdateOptions {
     
     /**
      * If true, allows the write to opt-out of document level validation. 
      * 
-     * On servers >= 3.2, the default is to not send a value. no 
-     * "bypassDocumentValidation" option is sent with the "findAndModify" command.
-     *
-     * On servers < 3.2, this option is ignored.
+     * This option is sent only if the caller explicitly provides a value. The default is to not send a value.
+     * For servers < 3.2, this option is ignored and not sent as document validation is not available.
      */
     bypassDocumentValidation: Optional<Boolean>;
     
     /**
+     * Specifies a collation.
+     *
+     * This option is sent only if the caller explicitly provides a value. The default is to not send a value.
+     * For servers < 3.4, the driver MUST raise an error if the caller explicitly provides a value.
+     *
+     * @see https://docs.mongodb.org/manual/reference/command/findAndModify/
+     */
+    collation: Optional<Document>;
+
+    /**
      * The maximum amount of time to allow the query to run.
      *
-     * @see http://docs.mongodb.org/manual/reference/command/findAndModify/
+     * @see https://docs.mongodb.org/manual/reference/command/findAndModify/
      */ 
     maxTimeMS: Optional<Int64>;
     
     /** 
      * Limits the fields to return for all matching documents.
      *
-     * @see http://docs.mongodb.org/manual/tutorial/project-fields-from-query-results
+     * This option is sent only if the caller explicitly provides a value. The default is to not send a value.
+     *
+     * Note: this option is mapped to the "fields" findAndModify field.
+     *
+     * @see https://docs.mongodb.org/manual/tutorial/project-fields-from-query-results
      */
     projection: Optional<Document>;
 
     /**
-     * When ReturnDocument.After, returns the updated or inserted document rather than the original.
-     * Defaults to ReturnDocument.Before.
+     * When ReturnDocument.After, returns the replaced or inserted document rather than the original.
+     *
+     * This option is sent only if the caller explicitly provides a value. The default is to not send a value.
      *
-     * @see http://docs.mongodb.org/manual/reference/command/findAndModify/
+     * Note: this option is mapped to the "new" findAndModify boolean field. ReturnDocument.Before is represents false, 
```

Comment:
Another place where we might want to say "command option" instead of "field".

Also, the "is" is a typo in "ReturnDocument.Before is represents false".